### ISSUE TITLE
Add alternative utility class for apache subnetutils

### ISSUE THREAD

**Issue by lsiepel** [openhab/openhab-core#3965]
We try to get rid of dependencies to apache libs per https://github.com/openhab/openhab-addons/issues/7722

we made great progress with https://github.com/openhab/openhab-core/pull/3738
Now there are 5 bindings that make use of subnetutils. I suggest a similar approach here, I can create an alternative util class in core to prevent code duplication. will be based on https://github.com/openhab/openhab-addons/pull/14430#issuecomment-1869707020 where the implementation was confirmed working as expected.

@openhab/core-maintainers any thoughts before I start working on it?

**Comment by J-N-K** [comment]
I don't think that's necessary. We need `org.apache.commons.net` anyway, so we can also use it in add-ons.

**Comment by lsiepel** [comment]
Oké, in that case I think  https://github.com/openhab/openhab-addons/issues/7722 can be closed, as these are the last dependencies in addons repo.
Full list is here https://github.com/openhab/openhab-addons/issues/7722#issuecomment-1432112689

@wborn  wdyt?

**Comment by wborn** [comment]
Maybe you can use `NetUtil` / `NetworkAddressService` instead? It has a `getAllInterfaceAddresses` method.
Add-ons should also respect the openHAB "Network Configuration" settings like not using IPv6 if it is disabled.
Currently we cannot remove commons-net because it is a dependency required for rfc2217 connections using nrjavaserial.

**Comment by lsiepel** [comment]
This issue is not about removing commons-net at **this** moment, its scope is much smaller. It is about reducing some dependencies to commons-net. I know we still depend on commons-net due to nrjavaserial, but why not reduce it as much as possible? 

I also see this as an opportunity to reduce and improve code that is used in multiple bindings that have about the same functionality. For instance, when i do a quick scan, it looks like none of the 5 involved bindings respect the openHAB "Network Configuration" settings.

**Comment by wborn** [comment]
> I also see this as an opportunity to reduce and improve code that is used in multiple bindings that have about the same functionality. For instance, when i do a quick scan, it looks like none of the 5 involved bindings respect the openHAB "Network Configuration" settings.

Yes that might be a nice improvement to fix multiple issues at once. :+1:

**Comment by lsiepel** [comment]
While working on this, i doubt if _'respecting' the openHAB "Network Configuration" settings_ is an improvement.

This configuration setting lets you choose a 'Primary Address'. How should one interpret the "Network Configuration" setting? 

My initial thought was that bindings should _only_ setup connections using this address. But when looking at multiple bindings                        i think this setting is much more limited. Main concerns when limiting bindings to this settings:
- none of the bindings respect this (breaking changes)
- what if you have vlan's or any other network segmentation with different devices on differnet networks.
- waht is the purpose of limiting bindings to use a network

So i guess this setting is only usefull for addons that expose a service and need to bind to a network address (like homekit etc). 

TL:DR i think if the related PR's are merged, this issue can also be closed. WDYT?
